### DISCUSSION TITLE
Prevent Extra HTTP request from Python?

### DISCUSSION THREAD

**Discussion by pweek** [seleniumbase/SeleniumBase#2664] [Q&A] [answered]
I've noticed an extra http request hitting the URL called when using uc=True. I see it occur right before the usual request when I check my web logs. It has user_agent "python-requests/2.31.0" but does not occur unless uc=True is used. Any way to prevent this?

from seleniumbase import Driver 
driver = Driver(uc=True)
driver.get(MYURL)
driver.quit()

**Comment by mdmintz** [comment, marked as answer]
That's covered in the UC Mode video tutorial: https://www.youtube.com/watch?v=5dMFI3e85ig

In UC Mode, `driver.get(url)` has been modified so that it first checks if UC Mode is needed for the website you're loading. If so, it calls `driver.uc_open_with_reconnect(url)`, and if not needed, it calls `driver.default_get(url)`, which is a copy of the original `driver.get(url)`.

In summary, to avoid that extra `requests` call, use either of the following when using UC Mode:
* `driver.default_get(url)`
* `driver.uc_open_with_reconnect(url)`.
